alf/bin/verify_checkpoint.py

```diff
@@ -0,0 +1,211 @@
+        policy_state, algorithm.get_initial_predict_state(batch_size),
+        time_step.is_first())
+    transformed_time_step, trans_state = algorithm.transform_timestep(
+        time_step, trans_state)
+    policy_step = algorithm.predict_step(transformed_time_step, policy_state,
+                                         epsilon_greedy)
+    return policy_step, trans_state
+
+
+def _run_steps(algorithm, env, nsteps, time_steps=[]):
+    """Run several steps of algorithm.
+
+    If ``time_steps`` is provided, will use them as the input.
+    Otherwise will obtain time steps through ``env``
+    """
+    policy_state = algorithm.get_initial_predict_state(env.batch_size)
+    trans_state = algorithm.get_initial_transform_state(env.batch_size)
+    policy_steps = []
+    run_env = not bool(time_steps)
+    if run_env:
+        time_step = common.get_initial_time_step(env)
+        time_steps.append(time_step)
+    else:
+        time_step = time_steps[0]
+    for i in range(1, nsteps + 1):
+        policy_step, trans_state = _step(
+            algorithm,
+            time_step,
+            policy_state,
+            trans_state,
+            epsilon_greedy=FLAGS.epsilon_greedy)
+        if run_env:
+            time_step = env.step(policy_step.output)
+            time_steps.append(time_step)
+        else:
+            time_step = time_steps[i]
+
+        policy_steps.append(policy_step)
+        policy_state = policy_step.state
+
+    return policy_steps, time_steps
+
+
+def _create_algorithm_and_env(root_dir):
+    """Create algorithm and env from config file."""
+    alf.reset_configs()
+    conf_file = common.get_conf_file()
+    try:
+        common.parse_conf_file(conf_file)
+    except Exception as e:
+        alf.close_env()
+        raise e
+    config = policy_trainer.TrainerConfig(root_dir=root_dir)
+
+    env = alf.get_env()
+    env.reset()
+    data_transformer = create_data_transformer(config.data_transformer_ctor,
+                                               env.observation_spec())
+    config.data_transformer = data_transformer
+
+    # keep compatibility with previous gin based config
+    common.set_global_env(env)
+    observation_spec = data_transformer.transformed_observation_spec
+    common.set_transformed_observation_spec(observation_spec)
+
+    algorithm_ctor = config.algorithm_ctor
+    algorithm = algorithm_ctor(
+        observation_spec=observation_spec,
+        action_spec=env.action_spec(),
+        reward_spec=env.reward_spec(),
+        config=config,
+        env=env)
+    return algorithm, env
+
+
+def main(_):
+    with tempfile.TemporaryDirectory() as root_dir:
+        FLAGS.root_dir = root_dir
+        conf_file = common.get_conf_file()
+        step_num = FLAGS.num_train_iterations
+        ckpt_dir = os.path.join(root_dir, 'ckpt')
+        algorithm1, env1 = _create_algorithm_and_env(root_dir)
+        for i in range(FLAGS.num_train_iterations):
```

Comment:
Better comment that here training is used to make sure the parameters of the algorithm actually changes to a certain state different from the initial state before checkpoint.

Reply:
Added.